akka-javasdk/src/main/scala/akka/javasdk/impl/serialization/JsonSerializer.scala

```diff
@@ -217,6 +219,29 @@ final class JsonSerializer(val objectMapper: ObjectMapper) {
       fromBytes(typeClass, bytesPayload)
   }
 
+  def exceptionFromBytes(exceptionPayload: BytesPayload): UserException = {
+    val exClass = exceptionClass(exceptionPayload)
+    fromBytes(exClass, exceptionPayload).asInstanceOf[UserException]
+  }
+
+  private def exceptionClass(exceptionPayload: BytesPayload): Class[_] = {
+    val className = stripJsonContentTypePrefix(exceptionPayload.contentType)
+    val exClass = reversedTypeHints.get(className)
+    if (exClass eq null) {
+      val loadedExceptionClass = Class.forName(className)
+      if (classOf[UserException].isAssignableFrom(loadedExceptionClass)) {
+        addToReversedCache(loadedExceptionClass, className)
+        loadedExceptionClass
+      } else {
+        throw new IllegalStateException(
+          "Loaded exception class [" + className +
+          "] is not a subclass of UserException, cannot deserialize it as a UserException.")
```

Comment:
```suggestion
          "Loaded class [" + className +
          "] is not a subclass of UserException, cannot deserialize it as a UserException.")
```

Very very unlikely to happen, but if it happens, it might not even be an exception at all.

Reply:
fixed